src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -25,4 +43,35 @@ public SearchIndexModel indexSnapshot(Snapshot snapshot, SearchIndexRequest sear
         List<Map<String, Object>> values = bigQueryPdao.getSnapshotTableData(snapshot, searchIndexRequest.getSql());
         return new SearchIndexModel();
     }
+    public SearchQueryResultModel querySnapshot(
+            SearchQueryRequest searchQueryRequest, List<String> indicesToQuery,
+            @Valid Integer offset, @Valid Integer limit
+    ) {
+        //todo: move to bean for configuration, make injectable
+        final RestClientBuilder builder = RestClient.builder( new HttpHost("localhost", 9200));
+        try (RestHighLevelClient client = new RestHighLevelClient(builder)) {
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            searchSourceBuilder.from(offset);
+            searchSourceBuilder.size(limit);
+            // see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wrapper-query.html
+            WrapperQueryBuilder wrapperQuery = QueryBuilders.wrapperQuery(searchQueryRequest.getQuery());
+            searchSourceBuilder.query(wrapperQuery);
+
+            SearchRequest searchRequest = new SearchRequest(indicesToQuery.toArray(new String[0]), searchSourceBuilder);
+
+            SearchResponse elasticResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+            SearchHits hits = elasticResponse.getHits();
+            List<String> response = new ArrayList<>();
+            for (SearchHit hit : hits) {
+                response.add(hit.getSourceAsString());
+            }
+            SearchQueryResultModel result = new SearchQueryResultModel();
+            //do we want to include info on the index/snapshot the response came from?
+            result.setResult(response.toString());
```

Comment:
What does the response look like?

Reply:
I'm curious about this too. I would think that this field should be valid JSON, and that this code would be building a JSON array for the response.

Reply:
for a result with two hits it looks like this:

```
{
  "result": "[{\"name\":\"John Doe\",\"age\":\"23\",\"background\":\"Born and brought up in California. Engineer by profession. Loves to cook\",\"nationality\":\"British\"}, {\"name\":\"test offset\",\"age\":\"28\",\"background\":\"Born and brought up in California. Engineer by profession. Loves to cook test offset\",\"nationality\":\"British\"}]"
}

result: [array of json blobs as string]

```

Reply:
Looks good enough for now, but I think we'll soon want to fix it so it's really JSON and not a string containing JSON. e.g.
```
{
  "result": [
        {"name":"John Doe","age":"23","background":"Born and brought up in California. Engineer by profession. Loves to cook","nationality":"British"}, 
        {"name":"test offset","age":"28","background":"Born and brought up in California. Engineer by profession. Loves to cook test offset","nationality":"British"}
    ]
}
```

Reply:
we would want to change the api to return json instead of a string in that case

Reply:
Figured I'd just do it in this pr, the response now looks like

```
{
  "result": [
    {
      "nationality": "British",
      "background": "Born and brought up in California. Engineer by profession. Loves to cook",
      "name": "John Doe",
      "age": "23"
    },
    {
      "nationality": "British",
      "background": "Born and brought up in California. Engineer by profession. Loves to cook test offset",
      "name": "test offset",
      "age": "28"
    }
  ]
}

```